### PR TITLE
Added support for setting and changing reminder due date and time

### PR DESCRIPTION
**Pull request by andylin2004**
A previous issue discussed about adding the intelligent date and time setter like in the reminders app (see https://github.com/DamascenoRafael/reminders-menubar/issues/17). This is a partial solution to that issue where users can change and set reminder due date and time manually (either below the textfield or via a popover that will apply changes once the user clicks out of the popover). English localization has also been applied for this.

### CODE REVIEW

File: reminders-menu-bar.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -541,7 +541,7 @@
 				COMBINE_HIDPI_IMAGES = YES;
 				DEAD_CODE_STRIPPING = YES;
 				DEVELOPMENT_ASSET_PATHS = "";
-				DEVELOPMENT_TEAM = 3398VKLH28;
+				DEVELOPMENT_TEAM = QW3PB89WQ9;
```
Comment (zydeico) [inline reminders-menu-bar.xcodeproj/project.pbxproj:544]: Delete changes in this file

**Review by zydeico — CHANGES_REQUESTED** [review]
LGTM but, delete changes in the **.pbxproj** file

**Review by zydeico — APPROVED** [review]

**Review by zydeico — APPROVED** [review]
LGTM

**Comment by DamascenoRafael** [comment]
Thanks for your contribution @andylin2004!
I know you've done a previous PR related to this, sorry I didn't look at this sooner, time is almost always short.

My macbook stopped turning on after an update, so I won't be able to show what I was doing and test your changes now. 
But as soon as I fix it I'll look into it.

Two small observations:
- Can you revert changes to Package.resolved? This file refers to a dependency needed by another feature.
- Can you install [SwiftLint](https://github.com/realm/SwiftLint)? After installing when building the project it will display warnings regarding code style. We don't need to solve all warnings, some can be ignored, but in general the idea is to try to follow some styles.

**Comment by andylin2004** [comment]
> Thanks for your contribution @andylin2004! I know you've done a previous PR related to this, sorry I didn't look at this sooner, time is almost always short.
> 
> My macbook stopped turning on after an update, so I won't be able to show what I was doing and test your changes now. But as soon as I fix it I'll look into it.
> 
> Two small observations:
> 
> * Can you revert changes to Package.resolved? This file refers to a dependency needed by another feature.
> * Can you install [SwiftLint](https://github.com/realm/SwiftLint)? After installing when building the project it will display warnings regarding code style. We don't need to solve all warnings, some can be ignored, but in general the idea is to try to follow some styles.

I have installed SwiftLint and adjusted my code to conform to the proper code style. Package.resolved isn't being tracked, so I think Xcode is autogenerating that.

**Comment by DamascenoRafael** [comment]
Hi, @andylin2004. I finally got my mac back.

The `Package.resolved` file is tracked, but you deleted it in your branch.
You can run the git command below to restore the master file to your branch:

```
git checkout origin/master reminders-menu-bar.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
```

I was working on an edit popover (similar to the one in Apple Reminders) allowing to edit the reminder's title, notes, date/time and priority.

Screenshot 2022-10-29 at 15 38 10

My idea is to combine this work (from the image above) with the possibility you created to add date/time when creating a reminder (image below). Sound good to you?

image


If it's ok I will merge your branch into a separate branch to combine these works and release this feature soon.

**Comment by andylin2004** [comment]
Sounds good! The initial version did have that, but I thought that was problematic because multiple popovers would then be needed.

**Comment by DamascenoRafael** [comment]
@andylin2004 can you please explain to me the advantage of using the `LegacyReminderTitleTextFieldView` instead of the default TextField in macOS versions below 12.0 (before macOS Monterey)?

**Comment by andylin2004** [comment]
The default textfield on macOS 11 will automatically create a reminder with the text in the textbook as the reminder name when the cursor clicks on the add date button (which is what we don't want to do, because people will probably adjust the due date multiple times).

```LegacyReminderTitleTextFieldView``` basically fixes this issue by expecting an enter key to be hit before the contents of the textbook is used to create a reminder

**Comment by DamascenoRafael** [comment]
Great! Thanks for the clarification. Didn't know about these behaviors on Big Sur.

**Pull request merged**